HW0/P4/P4.py

```diff
@@ -1,6 +1,27 @@
 import numpy as np
 import matplotlib.pyplot as plt
 from mpl_toolkits.mplot3d import Axes3D
+from numpy.linalg import inv
+
+
+# Function to calculate predictS
+def predictS(s_p):
```

Comment:
For all these functions, I see the rational behind not passing in the A, a, etc. parameters (it makes the signature much messier).

However, you're relying on the fact that these functions will be called only after the above are defined appropriately. It seems like asking for error/problems. I prefer to encapsulate everything as tightly as I can :)